plugin/library.py

```diff
@@ -45,22 +46,47 @@ def games(self):
                         path=Path(self.path).joinpath(
                             manifest['AppState']['installdir']),
                         id=manifest['AppState']['appid'],
-                        image_dir=Path(self.steam.path).joinpath(
-                            'appcache', 'librarycache'),
+                        image_dir=image_dir,
                     )
                 )
         return games
 
+class LibraryImageDir:
+    """
+    Caches the filesystem access of a library image directory
+    """
+
+    def __init__(self, image_dir: Union[str, Path]):
+        image_dir = Path(image_dir)
+        self.grid = image_dir.name == 'grid'
+        self._files_cache = {}
+        self._iterdir = image_dir.iterdir()
+
+    def get_image(self, id: str, type: str, sep='_') -> Path:
+        if self.grid:
+            # Grid images use short ID format
+            id = self.short_id()
```

Comment:
This method does not exist in this context

Reply:
Good catch!  I don't have any shortcuts so that didn't get tested by me running it.  I see a couple possible solutions:

* Pass in the whole LibraryItem instead of just the ID
* Pass in both the id and short_id
* Make ID it's own type that owns short_id()
* Make the short_id calculation more globally available and call it from here and LibraryItem

The first one gives lots of not-currently-needed info to this function and makes their dependency circular, but it actually does make some semantic sense and seems like that info could be needed if steam decides to switch the file naming convention maybe.  I'll update with that but am happy to implement one of the others instead if you prefer them.

Reply:
I think I got a very simple solution.... testing now.

Reply:
Hmmm, that one crossed my mind but I didn't love that it moved info about the image directory format into the LibraryItem class.  I've got the LibraryItem change ready if you want that instead.

```
diff --git a/plugin/library.py b/plugin/library.py
index f114787..af64f13 100644
--- a/plugin/library.py
+++ b/plugin/library.py
@@ -62,10 +62,11 @@ class LibraryImageDir:
         self._files_cache = {}
         self._iterdir = image_dir.iterdir()

-    def get_image(self, id: str, type: str, sep='_') -> Path:
+    def get_image(self, library_item: LibraryItem, type: str, sep='_') -> Path:
+        id = library_item.id
         if self.grid:
             # Grid images use short ID format
-            id = self.short_id()
+            id = library_item.short_id()

         prefix = f'{id}{sep}{type}'
         if prefix in self._files_cache:
@@ -113,7 +114,7 @@ class LibraryItem:
         webbrowser.open(self.uri())

     def get_image(self, type: str, sep='_') -> Path:
-        return self.image_dir.get_image(self.id, type, sep)
+        return self.image_dir.get_image(self, type, sep)
```